src/openms/include/OpenMS/ANALYSIS/ID/SiriusAdapterAlgorithm.h

```diff
@@ -0,0 +1,175 @@
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Oliver Alka $
+// $Authors: Oliver Alka $
+// --------------------------------------------------------------------------
+
+#pragma once 
+
+#include <OpenMS/DATASTRUCTURES/DefaultParamHandler.h>
+#include <OpenMS/KERNEL/MSSpectrum.h>
+#include <OpenMS/KERNEL/FeatureMap.h>
+#include <OpenMS/ANALYSIS/QUANTITATION/KDTreeFeatureMaps.h>
+#include <OpenMS/ANALYSIS/MAPMATCHING/FeatureMapping.h>
+
+#include <OpenMS/SYSTEM/File.h>
+#include <OpenMS/FORMAT/FeatureXMLFile.h>
+
+#include <QString>
+
+namespace OpenMS
+{
+  class OPENMS_DLLAPI SiriusAdapterAlgorithm : public DefaultParamHandler
+    {
+    public:
+      /// default constructor
+      SiriusAdapterAlgorithm();
+
+      /// Struct for temporary folder structure
+      struct SiriusTmpStruct
+      {
+        String tmp_dir;
+        String tmp_ms_file;
+        String tmp_out_dir;
+      };
+
+      /**
+        @brief Construct temporary folder structure for SIRIUS (SiriusTmpStruct)
+
+        @return SiriusTmpStruct
+      */
+      static SiriusAdapterAlgorithm::SiriusTmpStruct constructSiriusTmpStruct();
+
+      /**
+      @brief Checks if executable was povided 
+
+      @return Pair "path to executable" and "path to the working directory"
+
+      @param executable Path to the executable
+      */
+      static std::pair<String, String> checkSiriusExecutablePath(String& executable);
+
+
+      /**
+      @brief Preprocessing needed for SIRIUS
+
+      @return FeatureToMS2Indices
+            
+      Filter number of masstraces and perform feature mapping.
+
+      @param featureinfo: Path to featureXML
+      @param spectra: Input of MSExperiment with spectra information
+      @param fp_map_kd: KDTree used for query and match spectra with features
+      @param sirius_algo: Parameters needed for preprocessing
+      @param feature_mapping: Empty FeatureToMs2Indices
+      */
+      static void preprocessingSirius(const String& featureinfo,
+                                      MSExperiment& spectra,
+                                      std::vector<FeatureMap>& v_fp,
+                                      KDTreeFeatureMaps& fp_map_kd,
```

Comment:
The spectra could be const, but the v_fp and fp_map_kd not since they are filled in the function and in the end the pointer from the featrure map are referenced back to the vector of FeatureMap v_fp.

Reply:
maybe make this more clear in the `@param` so that its clear what is input and what is output